src/geometry/manifold/charted_space.lean

```diff
@@ -365,6 +366,86 @@ instance : order_top (structure_groupoid H) :=
   le_top := λ u f hf, by { split; exact dec_trivial },
   ..structure_groupoid.partial_order }
 
+/-- A groupoid is closed under restriction if it contains all restrictions of its element local
+homeomorphisms to open subsets of the source. -/
+class closed_under_restriction (G : structure_groupoid H) : Prop :=
+(closed_under_restriction : ∀ {e : local_homeomorph H H}, e ∈ G.members → ∀ (s : set H), is_open s →
+  (e : local_homeomorph H H).restr s ∈ G.members)
+
+/-- The trivial restriction-closed groupoid, containing only local homeomorphisms equivalent to the
+restriction of the identity to the various open subsets. -/
+def id_restr_groupoid : structure_groupoid H :=
```

Comment:
This looks like an instance of a more general construction: whenever you have a structure groupoid, you can construct a new groupoid by restricting the elements of the groupoid to open subsets. Then your `id_restr_groupoid` is this construction applied to `id_groupoid`. I am not sure it is worth developing the general construction, though, since I don't see other relevant applications. Keep it as an exercise for Friday? :)

Reply:
Actually, I think there is an application of the general construction.  It would be useful in constructing the groupoid whose elements are the restrictions of a subgroup of Homeo(H) to an open subset.  One needs this for affine manifolds, for example.  But let me do this an a future PR.

Reply:
ok, fine for me.